services/QuillLMS/client/app/bundles/Connect/components/activityHealth/activityHealth.tsx

```diff
@@ -0,0 +1,417 @@
+import * as React from 'react'
+import ReactTable from 'react-table'
+import { matchSorter } from 'match-sorter';
+import 'react-table/react-table.css';
+import request from 'request'
+import _ from 'underscore'
+import stripHtml from "string-strip-html"
+import { CSVLink } from 'react-csv'
+import { connect } from 'react-redux';
+
+import PromptHealth from './promptHealth'
+import NumberFilterInput from './numberFilterInput'
+
+import LoadingSpinner from '../shared/loading_indicator.jsx'
+import { sort, sortByList } from '../../../../modules/sortingMethods.js'
+import { FlagDropdown } from '../../../Shared/index'
+import { filterNumbers } from '../../../../modules/filteringMethods.js'
+import actions from '../../actions/activityHealth'
+
+const CONNECT_TOOL = "connect"
+const ACTIVITY_HEALTHS_URL = `${process.env.DEFAULT_URL}/api/v1/activities/activities_health.json`
+const ALL_FLAGS = "All Flags"
+const NO_DATA_FOUND_MESSAGE = "Activity Health data could not be found. Refresh to try again, or contact the engineering team."
+
+interface ActivityHealthProps {
+  dispatch: Function;
+  activityHealth: {
+    flag: string;
+  }
+}
+
+interface ActivityHealthState {
+  loadingTableData: boolean;
+  flag: string;
+  fetchedData: Array<Object>;
+  promptSearchInput: string;
+  dataToDownload: Array<Object>;
+}
+
+class ActivityHealth extends React.Component<ActivityHealthProps, ActivityHealthState> {
+
+  state = {
+    loadingTableData: true,
+    flag: '',
+    fetchedData: [],
+    promptSearchInput: "",
+    dataToDownload: []
+  };
+
+  componentDidMount() {
+    this.fetchActivityHealthData();
+  }
+
+  shouldComponentUpdate(nextProps, nextState){
+    return !(this.state === nextState)
+  }
```

Comment:
Why is this here?

Reply:
I need to stop the component from re-rendering when props are changed, and only re-render when state is changed, because I want the expand/collapse menus to stay open rather than close when, for example, the user goes to another tab and then clicks back to this tab.

Not sure if this is the correct approach but it did work for my purposes. Open to other suggestions!